pymarsys/connections.py

```diff
@@ -0,0 +1,143 @@
+from abc import ABC, abstractmethod
+import base64
+import datetime
+import hashlib
+import json
+import os
+import uuid
+
+import requests
+from aiohttp import ClientSession
+
+EMARSYS_URI = 'https://api.emarsys.net/'
+
+
+class ApiCallError(Exception):
+    pass
+
+
+class BaseConnection(ABC):
+    """
+    Any connection used to instantiate an Emarsys object or an object inherited
+     from BaseEndpoint should inherit from this class.
+    this class.
+    """
+    @abstractmethod
+    def __init__(self, username, secret, uri):
+        self.username = username
+        self.secret = secret
+        self.uri = uri
+
+    def build_authentication_variables(self):
```

Comment:
[personal taste]: i would rather define this function as a staticmethod because it is not especially dependent of many class attributes, and using a staticmethod forces oneself to define function parameters